src/esp/gfx/TextRenderer.cpp

```diff
@@ -0,0 +1,112 @@
+static void importFontResources() {
+  CORRADE_RESOURCE_INITIALIZE(FontResources)
+}
+
+namespace esp {
+namespace gfx {
+TextRenderer::TextRenderer(float aspectRatio,
+                           float fontSize,
+                           int cacheSize,
+                           const std::string& characters)
+    : cache_(Mn::Vector2i{cacheSize}) {
+  if (!Cr::Utility::Resource::hasGroup("fonts")) {
+    importFontResources();
+  }
+  CORRADE_ASSERT(
+      fontSize > 0,
+      "TextRenderer::TextRenderer: font size" << fontSize << "is illegal.", );
+  CORRADE_ASSERT(
+      cacheSize > 0,
+      "TextRenderer::TextRenderer: cache size" << cacheSize << "is illegal.", );
+
+  font_ = fontManager_.loadAndInstantiate("StbTrueTypeFont");
+  CORRADE_ASSERT(
+      font_,
+      "TextRenderer::TextRenderer: cannot load and instantiate the font", );
+
+  Cr::Utility::Resource rs("fonts");
+  std::string fontName = "SourceSansPro-SemiBold.ttf";
+  if (!font_->openData(rs.getRaw(fontName), fontSize)) {
+    Cr::Utility::Fatal{-1} << "TextRenderer::TextRenderer: The font" << fontName
+                           << "does not exist.";
+  }
+
+  font_->fillGlyphCache(cache_, characters);
+
+  updateAspectRatio(aspectRatio);
+}
+
+size_t TextRenderer::createRenderer(float onScreenCharacterSize,
+                                    size_t glyphCount,
+                                    Mn::Text::Alignment alignment) {
+  CORRADE_ASSERT(onScreenCharacterSize > 0,
+                 "TextRenderer::createRenderer: the character size"
+                     << onScreenCharacterSize << "is illegal.",
+                 0);
+
+  CORRADE_ASSERT(glyphCount > 0,
+                 "TextRenderer::createRenderer: the copacity" << glyphCount
+                                                              << "is illegal.",
+                 0);
+
+  text_.emplace_back(
+      std::move(Cr::Containers::Pointer<Mn::Text::Renderer2D>()));
+  size_t rendererId = text_.size() - 1;
+  text_[rendererId].reset(new Mn::Text::Renderer2D(
+      *font_, cache_, onScreenCharacterSize, alignment));
+  text_[rendererId]->reserve(glyphCount, Mn::GL::BufferUsage::DynamicDraw,
+                             Mn::GL::BufferUsage::StaticDraw);
+
+  return rendererId;
+}
+
+void TextRenderer::updateText(const std::string& text, size_t rendererId) {
+  CORRADE_ASSERT(rendererId < text_.size(),
+                 "TextRenderer::updateText: the rendererId"
+                     << rendererId << "is out of range.", );
+  text_[rendererId]->render(text);
+}
+
+TextRenderer& TextRenderer::draw(const Mn::Color3& color, size_t rendererId) {
+  CORRADE_ASSERT(
+      rendererId < text_.size(),
+      "TextRenderer::draw: the rendererId" << rendererId << "is out of range.",
+      *this);
+  textShader_
+      .setTransformationProjectionMatrix(
+          textProjection_ *
+          Mn::Matrix3::translation(
+              1.0f / textProjection_.rotationScaling().diagonal()))
```

Comment:
Again, this is copy+paste from your example program.

Reply:
> > Also I noticed something is rather confusing in the CI test.
> 
> It's because `WITH_STBTRUETYPEFONT` is enabled only if `BUILD_GUI_VIEWERS` is enabled ;)


I checked my code where I set it when BUILD_GUI_VIEWERS is enabled.
```
if(BUILD_GUI_VIEWERS)
    set(WITH_STBTRUETYPEFONT ON CACHE BOOL "WITH_STBTRUETYPEFONT" FORCE)
```
When I tried to compile it on linux, I used ```build.sh --headless```.  Wouldn't it set the BUILD_GUI_VIEWERS to false?